src/Test/Utilities/Portable/Diagnostics/DiagnosticExtensions.cs

```diff
@@ -104,6 +104,14 @@ public static TCompilation VerifyDiagnostics<TCompilation>(this TCompilation c,
             return c;
         }
 
+        public static TCompilation VerifyDiagnosticsAndEmitDiagnostics<TCompilation>(this TCompilation c, params DiagnosticDescription[] expected)
+            where TCompilation : Compilation
+        {
+            VerifyDiagnostics(c, expected);
```

Comment:
>VerifyDiagnostics(c, expected); [](start = 12, length = 31)

This feels redundant, but, if we are willing to "pay" for the time it takes to run tests, we can do that.

Reply:
We've been multiplying tests that check both `VerifyDiagnostics` and `VerifyEmitDiagnostics`, so this is just making it easier/shorter to write such tests.

Reply:
> We've been multiplying tests that check both VerifyDiagnostics and VerifyEmitDiagnostics, so this is just making it easier/shorter to write such tests.

I am not sure if that is really needed. I mean checking both, I was always suggesting to use one instead of the other. But I am fine with that as long as we are willing to wait longer for tests to complete.

---
In reply to: [455188087](https://github.com/dotnet/roslyn/pull/45898#discussion_r455188087) [](ancestors = 455188087)

Reply:
Okay. Switched to use `VerifyEmitDiagnostics` only.